src/Connector.sol

```diff
@@ -8,8 +8,14 @@ import {MemberlistLike} from "./token/memberlist.sol";
 // TODO: remove dependency on Messages.sol
 import {ConnectorMessages} from "src/Messages.sol";
 
-interface RouterLike {
-    function send(bytes memory message) external;
+interface GatewayLike {
+    function transfer(
+        uint64 poolId,
+        bytes16 trancheId,
+        ConnectorMessages.Domain destinationDomain,
```

Comment:
So this API remains under the same umbrella discussion we have ongoing. This API makes it seem like we can specify either `Domain.Centrifuge` or `Domain.EVM` but for the later we would also need a `ChainId` so this API doesn't really work as is. Note that this was the original API for the connector contract and we discussed in the PR adding outgoing message sending that we needed to make it type safe etc.